src/app.rs

```diff
@@ -484,14 +485,26 @@ where
             info!("Rustc git commit: {}", hash);
         }
 
+        let thread_count :usize = env::var("AMETHYST_NUM_THREADS")
+            .as_ref()
+            .map(String::as_str)
```

Comment:
The .parse.expect calls can be moved inside of the map clause, and in the unwrap_or clause you can directly put 0.

This will avoid a string allocation when the user does not use the env variable.

Reply:
Ah, interesting! Though I am not sure how to exactly move the `.parse.except` inside, could you point me in the right direction? I tried few guesses but end up with error akin to:

```

Reply:
.map(|s| s.as_str().parse().expect("...")).unwrap_or(0)

Reply:
Ah, awesome! Added in, and done, thank you very much!